src/output-json-alert.c

```diff
@@ -368,6 +377,14 @@ static int AlertJson(ThreadVars *tv, JsonAlertLogThread *aft, const Packet *p)
     json_object_clear(js);
     json_decref(js);
 
+    if (p->flags & PKT_HAS_TAG) {
+        MemBufferReset(aft->json_buffer);
+        json_t *packetjs = CreateJSONHeader((Packet *)p, 0, "packet");
```

Comment:
On Tue, Aug 23, 2016 at 9:15 AM, Victor Julien <notifications@github.com>
wrote:

> In src/output-json-alert.c
> <https://github.com/inliniac/suricata/pull/2196#discussion_r75887672>:
>
> > @@ -368,6 +377,14 @@ static int AlertJson(ThreadVars *tv, JsonAlertLogThread *aft, const Packet *p)
> >      json_object_clear(js);
> >      json_decref(js);
> >
> > +    if (p->flags & PKT_HAS_TAG) {
> > +        MemBufferReset(aft->json_buffer);
> > +        json_t *packetjs = CreateJSONHeader((Packet *)p, 0, "packet");
>
> should packetjs be checked for NULL here?
>
Yep, it should.